components/TARGET_PSA/TARGET_MBED_SPM/COMPONENT_SPE/handles_manager.c

```diff
@@ -47,35 +47,30 @@ psa_handle_t psa_hndl_mgr_handle_create(psa_handle_manager_t *handle_mgr, void *
     // Get active partition id - Needed for requester identification
     spm_partition_t *curr_part_ptr = get_active_partition();
     int32_t          current_pid   = ((curr_part_ptr != NULL) ? curr_part_ptr->partition_id : PSA_NSPE_IDENTIFIER);
-    uint32_t         expected      = UINT16_MAX;
-
-    // Avoid passing UINT16_MAX. Start again from 0 if reached.
-    // The reason for this is that we use the 16 upper bits to store the handle's index in the handles pool (for performance reasons)
-    core_util_atomic_cas_u32((uint32_t *)(&(handle_mgr->handle_generator)),
-                             &expected,
-                             PSA_HANDLE_MGR_INVALID_HANDLE
-                            );
 
     // Generate a new handle identifier
-    uint32_t tmp_handle = core_util_atomic_incr_u32(&(handle_mgr->handle_generator), 1);
-    uint32_t new_handle = PSA_HANDLE_MGR_INVALID_HANDLE;
-    uint32_t pool_ix    = 0;
+    uint32_t tmp_handle;
+    do {
+        tmp_handle = core_util_atomic_incr_u16(&(handle_mgr->handle_generator), 1);
+    } while (tmp_handle == PSA_HANDLE_MGR_INVALID_HANDLE);
```

Comment:
`PSA_HANDLE_MGR_INVALID_HANDLE` defined to be `uint16_t`.
i think it is better to compare to `PSA_NULL_HANDLE` instead.
`PSA_HANDLE_MGR_INVALID_HANDLE` is only used for static `handle_generator` initialisation.
```suggestion
    } while (tmp_handle == PSA_NULL_HANDLE);
```

Reply:
Yes, this one was debatable - all those constants are so coupled in their assumptions, it's a bit hard to tell which to use. I went for the `PSA_HANDLE_MGR_INVALID_HANDLE` as it is the direct 16-bit value of the `handle_generator`, despite having just popped it into a `uint32_t` ready for composition.

I think in principle `PSA_NULL_HANDLE` doesn't have to be `PSA_HANDLE_MGR_INVALID_HANDLE` zero-extended - if it wasn't my form is correct. (It might have been simpler if the invalidness was represented by a too-big value in the slot indicator, rather than a special generator count value, but maybe having the null value be 0 is convenient).

Reply:
`PSA_NULL_HANDLE` is part of PSA FF spec. The purpose of having it as it simplifies programming model. For example it allows us to call `psa_close()` regardless of `psa_connect()` status, which simplifies error handling logic.
PSA_HANDLE_MGR_INVALID_HANDLE is zero as it is initialises our handle_generator.
The connection between them is rather occasional.